R/fotmob_stats.R

```diff
@@ -153,39 +150,60 @@
         sprintf("Can't find season stats data. Failed with the following error:\n", resp$error)
       )
     }
-    stats <- .fotmob_extract_data_from_page_props(resp$result)$stats
+    stats <- resp$result$pageProps$stats
     seasons <- stats$seasonStatLinks$Name
 
     label <- sprintf("%ss", .upper1(team_or_player))
     valid_seasons <- setNames(seasons, seasons) %>%
-      purrr::imap_lgl(
-        ~label %in% stats[[.x]]$tabs
+      purrr::keep(
+        ~.x %in% stats$seasonStatLinks$Name
       ) %>%
-      purrr::keep(isTRUE) %>%
       names()
 
     extract_options <- function(season) {
-      all_stats_df <- stats[[season]][[sprintf("%ss", team_or_player)]] %>%
-        dplyr::distinct(.data[["header"]], .data[["name"]], .data[["fetchAllUrl"]])
-      season_id <- all_stats_df$fetchAllUrl %>% dirname() %>% basename() %>% unique()
-      stats_df <- all_stats_df %>% dplyr::distinct(.data[["header"]], .data[["name"]])
+
+      link <- stats$seasonStatLinks |>
```

Comment:
this needed major re-factoring due to changes made by fotmob

Reply:
ah, just realized i used the base pipe here! i can fix this next time i make a PR